### PR TITLE
[cli] Improve UX of graceful shutdown message

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Improve UX of graceful shutdown message
- base scripts : catch error code of action lifecycle and return it
- cha-lib : return error code based on http error code
- display better message when auth is required
- skip graceful mode when repo is mounted

### What issues does this PR fix or reference?
https://github.com/codenvy/codenvy/issues/1682

#### Changelog
Improves UX of graceful shutdown message.

#### Release Notes
We have made improvements to the CLI graceful shutdown behavior to catch additional errors, communicate more clearly when authentication is required for stop and skip the graceful mode when a repo is mounted.

#### Docs PR
n/a (improvement on UX, command is the same)

Change-Id: I1ad021426444b5b4d880df518512e53ef4828336
Signed-off-by: Florent BENOIT <fbenoit@codenvy.com>

### CODE REVIEW

**Review by bmicklea — APPROVED** [review]
tweaked release note entry

**Review by garagatyi — APPROVED** [review]
I'm not an expert in neither modern CLI code nor TS code we use, so can't do real technical review.
As for what I understand it looks good.

**Comment by benoitf** [comment]
@garagatyi I mentioned you if you wanted to check it was now working as you're expecting based the issue you reported.

**Review by TylerJewell — CHANGES_REQUESTED** [review]

File: dockerfiles/base/scripts/base/commands/cmd_stop.sh [hunk 1]
```diff
@@ -29,13 +29,23 @@ cmd_stop() {
   FORCE_STOP=false
   if [[ "$@" == *"--skip:graceful"* ]]; then
   	FORCE_STOP=true
+  elif local_repo; then
+    warning "Development mode [skip graceful stop]"
```
[the remark this answers is not in the document]
Comment (benoitf) [reply dockerfiles/base/scripts/base/commands/cmd_stop.sh]: In the issue ppl were asking for local repo mode
I'm also ok for local_assembly. I will add

File: dockerfiles/base/scripts/base/commands/cmd_stop.sh [hunk 2]
```diff
@@ -29,13 +29,23 @@ cmd_stop() {
...
   if server_is_booted $(get_server_container_id $CHE_CONTAINER_NAME); then 
     if [[ ${FORCE_STOP} = "false" ]]; then
       info "stop" "Stopping workspaces..."
-      if ! $(cmd_lifecycle action "graceful-stop" "$@" >> "${LOGS}" 2>&1 || false); then
-        error "We encountered an error -- see cli.log"
+      local GRACEFUL_STATUS_RESULT=0
+      cmd_lifecycle action "graceful-stop" "$@" >> "${LOGS}" 2>&1 || GRACEFUL_STATUS_RESULT=$?
+      # error on authentication (401 modulo 256 = 145)
+      if [[ ${GRACEFUL_STATUS_RESULT} -eq 145 ]]; then
+        error "Authentication failed on the system. Please provide --user and -password values or user --skip:graceful to bypass graceful stop."
```
[the remark this answers is not in the document]
Comment (benoitf) [reply dockerfiles/base/scripts/base/commands/cmd_stop.sh]: @TylerJewell in the issue it was requested to provide the argument for authentication (--user and --password) when auth is required

File: dockerfiles/base/scripts/base/commands/cmd_stop.sh [hunk 3]
```diff
@@ -29,13 +29,23 @@ cmd_stop() {
...
+        return 2;
+      elif [[ ${GRACEFUL_STATUS_RESULT} -ne 0 ]]; then
+        error "We encountered an error -- see $CHE_HOST_CONFIG/cli.log. Graceful stop can be skipped with --skip:graceful"
```
[the remark this answers is not in the document]
Comment (benoitf) [reply dockerfiles/base/scripts/base/commands/cmd_stop.sh]: shouldn't we add the hint on the fact that graceful stop can be skipped ?

[the remark this answers is not in the document]
Comment (TylerJewell) [reply dockerfiles/base/scripts/base/commands/cmd_stop.sh]: If you do it have it as (hint: `--skip:graceful` does not wait for workspace stop)

[the remark this answers is not in the document]
Comment (benoitf) [reply dockerfiles/base/scripts/base/commands/cmd_stop.sh]: ok

**Comment by benoitf** [comment]
@TylerJewell updated

**Review by TylerJewell — APPROVED** [review]

**Pull request merged**